lua/include/packet.lua

```diff
@@ -182,132 +134,278 @@ function pkt:enableTimestamps()
 	self.ol_flags = bit.bor(self.ol_flags, dpdk.PKT_TX_IEEE1588_TMST)
 end
 
--------------------------------------------------------------------------------------------
---- Return packet as XYZ
--------------------------------------------------------------------------------------------
 
-local etherPacketType = ffi.typeof("struct ethernet_packet*")
---- Retrieve an ethernet packet.
--- @return Packet in 'struct ethernet_packet' format
-function pkt:getEthernetPacket()
-	return etherPacketType(self.pkt.data)
-end
+----------------------------------------------------------------------------------
+--- Create new packet type
+----------------------------------------------------------------------------------
+
+--- Create struct and functions for a new packet
+-- @param args list of keywords (see makeStruct), followed by an optional table of namedDefaultArgs (see packet.fill)
+-- @return get return the 'cast to struct' function for this packet
+-- @see makeStruct
+-- @see _packetFill
+function packetCreate(...)
+	local args = {}
+	local defaultArgs = {}
+	for i, v in pairs{ select(1, ...) } do 
+		if type(v) == "table" then
+			defaultArgs = v
+		else
+			args[i] = v
+		end
+	end
+	
+	local packet = {}
+	packet.__index = packet
+	
+	packet.defaultArgs = defaultArgs
+
+	-- create struct and offer cast
+	local name, ctype = makeStruct(args)
+	packet.name = name
+
+	local get = function(self) return ctype(self.pkt.data) end
+
+	-- private/helper functions of the packet
+	packet.getHeaders = function(self) 
+		local headers = {} 
+		for i, v in ipairs(args) do 
+			headers[i] = _getHeader(self, v) 
+		end 
+		return headers 
+	end
 
-local arpPacketType = ffi.typeof("struct arp_packet*")
---- Retrieve an ARP packet.
--- @return Packet in 'struct arp_packet' format
-function pkt:getArpPacket()
-	return arpPacketType(self.pkt.data)
-end
+	-- public functions of the packet
+	packet.dump = function(self, bytes) _packetDump(self, bytes) end
```

Comment:
why isn't this, and the following definitions, just

    packet.dump = _packetDump

?